### PR TITLE
Add elFinder as filemanager for GrapesJs

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Just added elFinder as filemanager to GrapesJs. Build based on customization documentation https://grapesjs.com/docs/modules/Assets.html#customization

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Open the email
3. Insert image 
4. Check If elFinder works like in legacy  builder

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10902"></a>

### CODE REVIEW

File: plugins/GrapesJsBuilderBundle/Assets/library/js/builder.service.js [hunk 1]
```diff
@@ -260,15 +260,16 @@ export default class BuilderService {
    */
   getAssetManagerConf() {
     return {
-      assets: this.assets,
-      noAssets: Mautic.translate('grapesjsbuilder.assetManager.noAssets'),
-      upload: this.uploadPath,
```
Comment (adiux) [inline plugins/GrapesJsBuilderBundle/Assets/library/js/builder.service.js:265]: @kuzmany I guess we can then remove this parameter? And I think the actual PHP endpoint too.

**Review by adiux — CHANGES_REQUESTED** [review]
I think the current upload functionality can be removed?

File: plugins/GrapesJsBuilderBundle/Assets/library/js/builder.service.js (hunk 1 above)
Comment (kuzmany) [reply plugins/GrapesJsBuilderBundle/Assets/library/js/builder.service.js:265]: Yes, you're true

**Comment by kuzmany** [comment]
@adiux yes, we can remove it and avoid few security issues. I'll check it later.

**Review by adiux — CHANGES_REQUESTED** [review]
if we remove the not used endpoint I think this is good to go

**Review by jos0405 — APPROVED** [review]
Works great, I also tested email send, the images appear.
Good to go.

**Review by deborahsalves — APPROVED** [review]
image
It works as expected. Approved

**Comment by RCheesley** [comment]
@all-contributors please add @deborahsalves  for UserTesting

**Comment by jos0405** [comment]
> if we remove the not used endpoint I think this is good to go

@kuzmany can you plz do that so we can merge this amazing PR?

**Comment by kuzmany** [comment]
This PR need https://github.com/mautic/mautic/pull/10847 to access of editor of object with `Mautic.builder` 
Then I am able to fix close elFinder and re-open it - at the moment does not work. We have internally fixed it , but in the case https://github.com/mautic/mautic/pull/10847 is not gonna finish and merged it, I can push our temporary solution.

**Comment by adiux** [comment]
@kuzmany The main problem of https://github.com/mautic/mautic/pull/10847 is fixed. But I'm still stuck with the new problem that when the email is saved, the editor does not open again. So something happens on this save request that I don't quite understand yet. There is so much code involved to reverse engineer. 

I guess it is related to some functionality related to the old builder. If anybody has some insights that would be great!

**Comment by adiux** [comment]
@kuzmany I dont think it is a great idea to rely on the the global `Mautic.builder`. We should try to avoid it as it creates more of these intransparent dependencies like the one I just face with the save button. Maybe there is a solution that does not rely on global variables?

But if I look at the code you are not using `Mautic.builder` anyway?

**Comment by kuzmany** [comment]
Not ready to commit yet. Still need push some commits

**Comment by adiux** [comment]
@RCheesley @kuzmany I don't think there is a dependency to https://github.com/mautic/mautic/pull/10847

We should also remove the current file upload endpoint in PHP if possible.

**Comment by adiux** [comment]
@kuzmany this would be a great feature for the v5?

**Comment by kuzmany** [comment]
Close favor to https://github.com/mautic/mautic/pull/11727

**Pull request closed without merging**